### PR TITLE
Update chains to 0.9.0

### PR DESCRIPTION
**Pull request by caugello**

### CODE REVIEW

File: components/build/tekton-chains/chains-controller-deployment.yaml [hunk 1]
```diff
@@ -8,12 +8,9 @@ spec:
     spec:
       containers:
       - name: tekton-chains-controller
-        # Use a nightly-build image because the current latest version, 0.8.0, contains
-        # several bugs that make Chains unusable for our use case. Once version 0.9.0
-        # is published, change this reference to that. Contract team fork may be useful
+        # Contract team fork may be useful
         # in future for testing features not yet upstream:
         # image: ghcr.io/hacbs-contract/chains/controller@sha256...
-        image: gcr.io/tekton-nightly/github.com/tektoncd/chains/cmd/controller@sha256:9b4bef2e9d617dddfc0243b78db2ce8b0f176460bc408bd28e5b31cba79bdece
```
Comment (simonbaird) [inline components/build/tekton-chains/chains-controller-deployment.yaml:16]: Not a blocker, but keeping the url commented out could save some time in future, e.g.:

    # Nightly builds of chains if required:
    #image: gcr.io/tekton-nightly/github.com/tektoncd/chains/cmd/controller@sha256...

File: components/build/tekton-chains/chains-controller-deployment.yaml [hunk 2]
```diff
@@ -8,12 +8,9 @@ spec:
     spec:
       containers:
       - name: tekton-chains-controller
-        # Use a nightly-build image because the current latest version, 0.8.0, contains
-        # several bugs that make Chains unusable for our use case. Once version 0.9.0
-        # is published, change this reference to that. Contract team fork may be useful
+        # Contract team fork may be useful
         # in future for testing features not yet upstream:
         # image: ghcr.io/hacbs-contract/chains/controller@sha256...
```
Comment (simonbaird) [inline components/build/tekton-chains/chains-controller-deployment.yaml:13]:
```suggestion
        # image: ghcr.io/hacbs-contract/chains/controller@sha256...
       -# To use a nightly build of chains if required:
       +# image: gcr.io/tekton-nightly/github.com/tektoncd/chains/cmd/controller@sha256...
```

Comment (lcarva) [reply components/build/tekton-chains/chains-controller-deployment.yaml:13]: +1 for including both image examples.

**Review by simonbaird — APPROVED** [review]

**Comment by simonbaird** [comment]
Since I merged another patch, now this needs another rebase and another CI run..

**Review by simonbaird — APPROVED** [review]

File: components/build/tekton-chains/chains-controller-deployment.yaml [hunk 3]
```diff
@@ -8,12 +8,11 @@ spec:
...
-        image: gcr.io/tekton-nightly/github.com/tektoncd/chains/cmd/controller@sha256:9b4bef2e9d617dddfc0243b78db2ce8b0f176460bc408bd28e5b31cba79bdece
+       -# To use a nightly build of chains if required:
+       +# image: gcr.io/tekton-nightly/github.com/tektoncd/chains/cmd/controller@sha256...
```
Comment (zregvart) [inline components/build/tekton-chains/chains-controller-deployment.yaml:15]:
```suggestion
        # To use a nightly build of chains if required:
        # image: gcr.io/tekton-nightly/github.com/tektoncd/chains/cmd/controller@sha256...
```

**Review by zregvart — COMMENTED** [review]
Seems that the waiting on `tekton-chains` namespace, i.e. this bit:
https://github.com/redhat-appstudio/infra-deployments/blob/d217d552a79a8eba0172ccb103cbe22783da1809/hack/preview.sh#L108-L112
Takes a bit over [107 minutes](https://www.wolframalpha.com/input?i=2147*3+seconds) counting by the 2147 dots each 3 seconds apart in the [log](https://storage.googleapis.com/origin-ci-test/pr-logs/pull/redhat-appstudio_infra-deployments/348/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-deployment/1524879397176217600/build-log.txt). Which is odd because the job succeeded [twice before](https://prow.ci.openshift.org/pr-history/?org=redhat-appstudio&repo=infra-deployments&pr=348).

This is from the ArgoCD [server logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_infra-deployments/348/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-deployment/1524879397176217600/artifacts/appstudio-e2e-deployment/gather-extra/artifacts/pods/openshift-gitops_openshift-gitops-repo-server-559f89bd96-wtwk7_argocd-repo-server.log):

```
time="2022-05-12T22:36:50Z" level=error msg="`kustomize build /tmp/_argocd-repo/aff0e4b4-a2e1-4c4c-8415-063f34d3db79/components/build` failed exit status 1: Error: accumulating resources: accumulation err='accumulating resources from 'tekton-chains/': '/tmp/_argocd-repo/aff0e4b4-a2e1-4c4c-8415-063f34d3db79/components/build/tekton-chains' must resolve to a file': recursed accumulation of path '/tmp/_argocd-repo/aff0e4b4-a2e1-4c4c-8415-063f34d3db79/components/build/tekton-chains': trouble configuring builtin PatchStrategicMergeTransformer with config: `\npaths:\n- chains-config.yaml\n- security-context-fix.yaml\n- chains-controller-deployment.yaml\n`: yaml: line 8: did not find expected key" execID=34a7d
time="2022-05-12T22:36:50Z" level=info msg=Trace args="[kustomize build /tmp/_argocd-repo/aff0e4b4-a2e1-4c4c-8415-063f34d3db79/components/build]" dir= operation_name="exec kustomize" time_ms=16057.316400000002
time="2022-05-12T22:36:50Z" level=error msg="finished unary call with code Unknown" error="`kustomize build .components/build` failed exit status 1: Error: accumulating resources: accumulation err='accumulating resources from 'tekton-chains/': '.components/build/tekton-chains' must resolve to a file': recursed accumulation of path '.components/build/tekton-chains': trouble configuring builtin PatchStrategicMergeTransformer with config: `\npaths:\n- chains-config.yaml\n- security-context-fix.yaml\n- chains-controller-deployment.yaml\n`: yaml: line 8: did not find expected key" grpc.code=Unknown grpc.method=GenerateManifest grpc.request.deadline="2022-05-12T22:37:33Z" grpc.service=repository.RepoServerService grpc.start_time="2022-05-12T22:36:33Z" grpc.time_ms=16698.51 span.kind=server system=grpc
```

Which is detecting an issue in `chains-controller-deployment.yaml`:

```shell
$ yq components/build/tekton-chains/chains-controller-deployment.yaml 
Error: bad file 'components/build/tekton-chains/chains-controller-deployment.yaml': yaml: line 8: did not find expected key
```

So the typo (suggested a change) caused the e2e to fail.

**Pull request merged**